sdcm/remote/libssh2_client/__init__.py

```diff
@@ -440,25 +440,31 @@ def _process_output(  # pylint: disable=too-many-arguments, too-many-branches
             if perf_counter() > end_time:
                 reader.stop()
                 return False
-            if stdout_stream is not None:
+
+            data_processed = False
+            if stdout_stream is not None and reader.stdout.qsize():
                 try:
                     stdout = reader.stdout.get(timeout=timeout_read_data_chunk)
                     data = stdout.decode(encoding, errors='ignore') + '\n'
                     stdout_stream.write(data)
                     for watcher in watchers:
                         watcher.submit_line(data)
+                    data_processed = True
                 except Exception:  # pylint: disable=broad-except  # noqa: BLE001
                     pass
-            if stderr_stream is not None:
+            if stderr_stream is not None and reader.stderr.qsize():
```

Comment:
This is redundant change, the next line does the same check.

Reply:
true, I didn't notice it's a bit different in stderr handler.
Removed duplication from the inside, making one indent less.